base/steer/FairRun.h

```diff
@@ -205,6 +207,8 @@ class FairRun : public TNamed
     FairRun& operator=(const FairRun&) { return *this; }
     /** Number of Tasks added*/
     Int_t fNTasks;
+    /** MC Event Header */
+    std::unique_ptr<FairEventHeader> fEvtHeader;   //!
```

Comment:
I really appreciate, that you're making this private.

But maybe let's do things step by step and keep it protected for the time being and fix up all those users that can't handle a `unique_ptr`? The major one is probably FillEventHeader(). I had #1255 in my private tree anyway and thought it's now a good time for it. It should make this much simpler. So maybe you might want to wait for it getting merged and then rebase?

Reply:
I would rather like to keep it private. If I make it protected (, which I tried in the very first beginning), lots of places would need to be revised because of of the naming. If other derived class need to use this private pointer, just call the function `GetEventHeader()`

Reply:
How about changing it to 
```cpp
fEvtHeader = GetEventHeader();
```

Reply:
Thinking again about it…
```cpp
fEvtHeader = dynamic_cast<FairEventHeader*>(fRootManager->GetObject("EventHeader."));
```

I think, this assumed (in the current version), that `GetObject` returns a fresh/new object. Because currently the destructor of `FairRun` deletes it.

So what about replacing this just with `SetEventHeader(…)`? Then the unique_ptr will do the `delete` (again).

Reply:
Yeah, I would also do this. Event header should be directly obtained from `FairRun` every time for the consistency.